AirCasting/SessionViews/SessionHeaderView.swift

```diff
@@ -113,7 +116,10 @@ private extension SessionHeaderView {
             }
             
             Button {
-                // action here
+                DispatchQueue.global(qos: .background).async {
+                    networkChecker.monitorNetwork()
```

Comment:
I'd suggest creating `NetworkChecker` way earlier in the apps lifetime. Reason is it needs some time to evaluate stuff, it might give you fake results if you start monitoring here and check it on the very next instruction. I think the best way would be to instantiate it in `RootAppView` and pass down using environment objects